cardano-cli/src/Cardano/CLI/Shelley/Run/Node.hs

```diff
@@ -244,3 +271,25 @@ readColdVerificationKeyOrFile coldVerKeyOrFile =
         , FromSomeType (AsVerificationKey AsGenesisDelegateKey) castVerificationKey
         ]
         fp
+
+#ifdef UNIX
+writeTextEnvelopeFileWithOwnerPermissions
+  :: HasTextEnvelope a
+  => FilePath
+  -> Maybe TextEnvelopeDescr
+  -> a
+  -> ExceptT ShelleyNodeCmdError IO ()
+writeTextEnvelopeFileWithOwnerPermissions targetPath mbDescr a = do
+  let content = textEnvelopeToJSON mbDescr a
+      (targetDir, targetFile) = splitFileName targetPath
+  liftIO $ Exception.bracketOnError
+             (openTempFile targetDir $ targetFile <.> "tmp")
+             (\(tmpPath, fHandle) -> do
+               hClose fHandle >> removeFile tmpPath
+               return $ ShelleyNodeCmdVrfSigningKeyCreationError targetPath tmpPath)
+             (\(tmpPath, fHandle) -> do
+                 LBS.hPut fHandle content
+                 hClose fHandle
+                 renameFile tmpPath targetPath
+                 setFileMode targetPath ownerModes)
```

Comment:
The `setFileMode` should be unnecessary. The docs for `openTempFile` says:

> The file is created with permissions such that only the current user can read/write it.

Which is exactly what we want.